src/Desktop/UIAutomation/A11yAutomation.cs

```diff
@@ -147,6 +155,35 @@ public static IEnumerable<DesktopElement> ElementsFromProcessId(int pid, Desktop
             return elements;
         }
 
+        internal static DesktopElement GetFocusedElement()
+        {
+            IUIAutomation uiAutomation = GetDefaultInstance().UIAutomation;
+            IUIAutomationElement focusedElement = uiAutomation.GetFocusedElement();
+            return new DesktopElement(focusedElement, keepElement: true, setMembers: true);
+        }
+
+        internal static DesktopElement GetDepthFirstLastLeafControlElement(DesktopElement rootElement)
+        {
+            var walker = GetDefaultInstance().GetTreeWalker(TreeViewMode.Control);
+            try
+            {
+                IUIAutomationElement leafElement = (IUIAutomationElement)rootElement.PlatformObject;
+                for (IUIAutomationElement currentElement = walker.GetLastChildElement(leafElement); currentElement != null; currentElement = walker.GetLastChildElement(leafElement))
+                {
+                    Marshal.ReleaseComObject(leafElement);
+                    leafElement = currentElement;
+                }
+
+                return leafElement is null
+                    ? rootElement
+                    : new DesktopElement(leafElement, keepElement: true, setMembers: true);
+            }
+            finally
+            {
+                Marshal.ReleaseComObject(walker);
+            }
+        }
+
```

Comment:
`GetFocusedElement` and `GetDepthFirstLastLeafControlElement` exist purely for the integration tests. As such, I'd argue that they probably should exist in the unit tests

Reply:
Yep, moving this.

Reply:
Added in latest iteration